packages/flutter/test/material/scaffold_test.dart

```diff
@@ -998,6 +998,56 @@ void main() {
       semantics.dispose();
     });
 
+    testWidgets('Drawer state query correctly', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        MaterialApp(
+          home: SafeArea(
+            left: false,
+            top: true,
+            right: false,
+            bottom: false,
+            child: Scaffold(
+              endDrawer: const Drawer(
+                child: Text('endDrawer'),
+              ),
+              drawer: const Drawer(
+                child: Text('drawer'),
+              ),
+              body: const Text('scaffold body'),
+              appBar: AppBar(
+                centerTitle: true,
+                title: const Text('Title')
+              ),
+            ),
+          ),
+        ),
+      );
+
+      // Open Drawer, tap on end drawer, which closes the drawer, but does
+      // not open the drawer.
+      await tester.tap(find.byType(IconButton).first);
```

Comment:
The value of this Finder is a bit mysterious. It would help to bind the expression to a var whose name explains what it is.

Likewise for the other IconButton finders used throughout.